File: backend/app/schemas/billing.py
from __future__ import annotations
from pydantic import BaseModel
from typing import Optional


class MonthlyCostItem(BaseModel):
    year: int
    month: int
    period_start: str
    period_end: str
    total_cost: float
    unit: str
    label: str

    model_config = {"from_attributes": True}


class MonthlyCostListResponse(BaseModel):
    data: list[MonthlyCostItem]
    count: int
    currency: str = "USD"


class ServiceCostItem(BaseModel):
    service_name: str
    cost: float
    unit: str
    percentage: float

    model_config = {"from_attributes": True}


class ServiceBreakdownResponse(BaseModel):
    year: int
    month: int
    label: str
    total_cost: float
    services: list[ServiceCostItem]
    count: int


class SummaryMonthItem(BaseModel):
    label: str
    cost: float
    unit: str


class SummaryYTD(BaseModel):
    year: int
    cost: float
    unit: str


class SummaryResponse(BaseModel):
    current_month: SummaryMonthItem
    last_month: SummaryMonthItem
    ytd: SummaryYTD
    mom_change_pct: Optional[float]

File: frontend/src/types/billing.ts
export interface MonthlyCostItem {
  year: number
  month: number
  period_start: string
  period_end: string
  total_cost: number
  unit: string
  label: string
}

export interface MonthlyCostListResponse {
  data: MonthlyCostItem[]
  count: number
  currency: string
}

export interface ServiceCostItem {
  service_name: string
  cost: number
  unit: string
  percentage: number
}

export interface ServiceBreakdownResponse {
  year: number
  month: number
  label: string
  total_cost: number
  services: ServiceCostItem[]
  count: number
}

export interface SummaryMonthItem {
  label: string
  cost: number
  unit: string
}

export interface SummaryResponse {
  current_month: SummaryMonthItem
  last_month: SummaryMonthItem
  ytd: { year: number; cost: number; unit: string }
  mom_change_pct: number | null
}

export interface NoteItem {
  id: number
  year: number
  month: number
  service_name: string
  note: string
  note_date: string
  resource_id: string | null
  resource_name: string | null
  filter_name: string | null
  cloud: string
  cloud_account: string
  aws_profile: string
  created_at: string
}

export interface SyncStatusResponse {
  sync_id: number
  status: string
  started_at: string
  finished_at: string | null
  months_synced: number
  cloud: string
  cloud_account: string
  aws_profile: string | null
  aws_region: string | null
  error_message: string | null
}

export interface CloudAccount {
  cloud: string
  cloud_account: string
}
